dhcpv4/client.go

```diff
@@ -118,6 +118,10 @@ func MakeBroadcastSocket(ifname string) (int, error) {
 // MakeListeningSocket creates a listening socket on 0.0.0.0 for the DHCP client
 // port and returns it.
 func MakeListeningSocket(ifname string) (int, error) {
+	return makeListeningSocketWithCustomPort(ifname, ClientPort)
+}
+
+func makeListeningSocketWithCustomPort(ifname string, port int) (int, error) {
```

Comment:
do we need this? Can't we just call `c.getLocalUDPAddr()` inside MakeListeningSocket` and use that port? The port will be already set accordingly

Reply:
We don't currently have access to the client inside `MakeListeningSocket()`, so we would have to change that, which I can do without problems (and leave the current function untouched?)

Reply:
no, you're right, I didn't realize that. Better to leave the interface as it is. I am not fond of adding new functions, but this one is OK